packages/opensrp-server-service/src/serviceClass.ts

```diff
@@ -35,16 +35,20 @@ export function getDefaultHeaders(
  * @param {object} _ - signal object that allows you to communicate with a DOM request
  * @param {string} accessToken - the access token
  * @param {string} method - the HTTP method
+ * @param {object} data - data to be used for payload
  * @returns {Object} the payload
  */
 export function getFetchOptions(
   _: AbortSignal,
   accessToken: string,
-  method: HTTPMethod
-): { headers: HeadersInit; method: HTTPMethod } {
+  method: HTTPMethod,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: any
```

Comment:
Can we use `Dictionary` type here?

Reply:
Maybe even a generic type that defaults to `Dictionary`?